src/pages/AccountDeletion.js

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useState, useContext } from "react";
+import { UserContext } from "../contexts/UserContext";
+import InfoPage from "./info_page_shared/InfoPage";
+import Header from "./info_page_shared/Header";
+import Heading from "./info_page_shared/Heading";
+import Main from "./info_page_shared/Main";
+import Footer from "./info_page_shared/Footer";
+import LoadingAnimation from "../components/LoadingAnimation";
+import redirect from "../utils/routing/routing";
+import SessionStorage from "../assorted/SessionStorage";
+import { APP_DOMAIN } from "../appConstants";
+
+export default function AcountDeletion({ api }) {
+  const user = useContext(UserContext);
+  const [headingMsg, setHeadingMsg] = useState("We are deleting your account");
+  const [errorOcurred, setErrorOcurred] = useState();
+  const [isAccountDeleted, setIsAccountDeleted] = useState();
+  useEffect(() => {
+    console.log(user);
+    if (SessionStorage.hasUserConfirmationForAccountDeletion()) {
+      setErrorOcurred(false);
+      setIsAccountDeleted(false);
+      SessionStorage.setUserConfirmationForAccountDeletion(false);
+      api.deleteUser(
+        () => {
+          setIsAccountDeleted(true);
+          setHeadingMsg("✅ Your account has been successfully deleted!");
+          setTimeout(() => {
+            user.logoutMethod();
+          }, [5000]);
+        },
+        (error) => {
+          setIsAccountDeleted(false);
+          setErrorOcurred(true);
+          setHeadingMsg("❌ An error has occurred when deleting your account.");
+        },
+      );
+    } else {
+      redirect(APP_DOMAIN);
+    }
+  }, []);
+  // In case the user went to the path by mistake
```

Comment:
i don't get this comment. does it refer to the next line?

Reply:
Yes, not sure why it was down there!